tools/realsense-viewer/auto_updater/update_handler.cpp

```diff
@@ -0,0 +1,271 @@
+// License: Apache 2.0. See LICENSE file in root directory.
+// Copyright(c) 2017 Intel Corporation. All Rights Reserved.
+#include <fstream>
+#include <unordered_set>
+#include <algorithm>
+
+#include "json.hpp"
+#include "update_handler.h"
+#include "http_downloader.h"
+
+namespace rs2
+{
+    using json = nlohmann::json;
+
+    long long update_handler::check_for_updates(const component_part_type cp, const update_policy_type up)
+    {
+        // Load server versions info on first access
+        if (!_server_versions_loaded)
+        {
+            if (load_server_versions())
+            {
+                _server_versions_loaded = true;
+            }
+            else
+            {
+                return 0; // Cannot download server file
+            }
+        }
+
+        auto platform = get_platform();
+
+        // Look for the required version
+        auto res = std::find_if(_server_versions_vec.begin(), _server_versions_vec.end(),
+            [cp, up, platform](server_version_type ver) {return ((ver.component == cp) && (ver.policy == up) && (ver.platform == platform)); });
+
+        if (res != _server_versions_vec.end())
+        {
+            return (*res).version;
+        }
+
+        return 0; // Nothing found
+    }
+
+    bool update_handler::get_download_link(const component_part_type cp, const long long version, std::string& dl_link)
+    {
+        // Load server versions info on first access
+        if (!_server_versions_loaded)
```

Comment:
extract this logic to private init() function?

Reply:
The "load_server_versions" is the init function.
I did not want to call it on each entry,  only if it the first time so I wrapped it with a flag.
Do you mean you suggest renaming it to init and do the flag logic inside?

Reply:
Fixed